src/go/k8s/pkg/resources/configmap.go

```diff
@@ -159,75 +135,43 @@ func (r *ConfigMapResource) Obj() (k8sclient.Object, error) {
 	return cm, nil
 }
 
-func (r *ConfigMapResource) createConfiguration() interface{} {
-	serviceAddress := r.pandaCluster.Name + "." + r.pandaCluster.Namespace + ".svc.cluster.local"
+func (r *ConfigMapResource) createConfiguration() *config.Config {
+	serviceAddress := r.svc.HeadlessServiceFQDN()
 	cfgRpk := config.Default()
 
 	c := r.pandaCluster.Spec.Configuration
 	cr := cfgRpk.Redpanda
```

Comment:
Should this be `&cfgRpk.Redpanda`? We're making a copy of `.Redpanda` and return `cfgRpk`.

Reply:
You are right :) Fixed.